impeller/aiks/canvas.cc

```diff
@@ -386,7 +386,12 @@ void Canvas::DrawVertices(const std::shared_ptr<VerticesGeometry>& vertices,
   entity.SetStencilDepth(GetStencilDepth());
   entity.SetBlendMode(paint.blend_mode);
 
-  if (!vertices->HasVertexColors()) {
+  // If there are no vertex color or texture coordinates. Or if there
+  // are vertex coordinates then only if the contents are an image.
+  if (!vertices->HasVertexColors() &&
+      (!vertices->HasTextureCoordinates() ||
+       (vertices->HasTextureCoordinates() &&
+        paint.color_source_type == Paint::ColorSourceType::kImage))) {
```

Comment:
nit: I think this and the `if` below would read a little clearer if there was a method on `Vertices` to help sort these things out.

Reply:
There's not really a good place to put this. it depends on the color source type which can't be used outside of aiks, and the vertices are a geometry object defined in entity. It can't go on the contents because it isn't created yet.

Reply:
moved this to a static helper function